### ISSUE TITLE
Login Issues

### ISSUE THREAD

**Issue by mstern07** [dweeves/magmi-git#354]
I just downloaded latest version of magmi and placed it on my root. I get the login prompt and enter the username/password "magmi" (without quotes) and the same popup comes up again. I think tryied creating magmi.ini with my database credentials and am still unable to login using my magento admin login. Any ideas?

Thanks,
Mike

**Comment by liam-wiltshire** [comment]
Hi @mstern07,

What is your server setup? Version of apache, PHP, Magento?

**Comment by mstern07** [comment]
Apache 2.2.29
PHP 5.3.24
Magento: latest version (downloaded/installed 2 days ago)

Thanks.

**Comment by aziwaan** [comment]
This sounds similar to what Im experiencing.  Could it be to do with the Apache version and a htpasswd bug?  i havnt looked to closely at the latest changes but setting up htpasswd on that version of apache has given me problems in the past.

**Comment by gregrobbins** [comment]
Same issue here, does not work with 'magmi' username and password.

**Comment by dweeves** [comment]
The latest master accepts magmi/magmi IF AND ONLY IF magmi was not already configured for imports.
Otherwise , use your standard magento admin credentials.

**Comment by MitchellBouwman** [comment]
Have you found out already @mstern07? 
Having the same issue.. tried 'magmi' - 'magmi' and username and password of my magento admin account.

**Comment by mstern07** [comment]
I tried everything magmi/magmi, magento store login, sql database login and nothing worked. I ended up installing an older version of magmi.

Mike

**Comment by MitchellBouwman** [comment]
Thanks alot @mstern07, can't figure it out also. Will try an older version!

**Comment by MitchellBouwman** [comment]
Yep logged in instantly now with an older version. Thanks @mstern07!

**Comment by BenButterfield** [comment]
Similar problem. My Magento install is already behind htpasswd whilst in development and I wonder if that's the conflict. I also went with the older version.

**Comment by Josch-** [comment]
Hi,

same problem here - no login possible. Fresh installation of magmi in a new folder, so no magmi.ini is present - login with "magmi"/"magmi" should work - nothing. With a configured magmi.ini and admin username and password - nothing. Everytime the same popup for username and password!

PHP version 5.4.45
Apache version ???

Downgraded too to 0.7.21 :-(

Any help would be appreciated!

Best wishes,

Josch

**Comment by apzeiro** [comment]
same problem here.

**Comment by sarog** [comment]
I'm also experiencing this issue on the latest source version of Magmio. I tried adding the 3 lines (from the bottom of README.MD) to my .htaccess file without any luck either. I also attempted to create the magmi.ini manually with my database credentials: still no go. Of course, I tried logging in with magmi/magmi and no dice. My Magento admin credentials don't work either.

Apache 2.4.18, php-fpm 5.6.19, Magento 1.9.2.4

**Comment by sarog** [comment]
Wanted to add: no errors appear in the Apache log files.

**Comment by dweeves** [comment]
use the magento admin credentials if you already have configured an older
magmi on your install.

it should work.

2016-04-15 0:29 GMT+02:00 Saro <notifications@github.com>:

**Comment by sarog** [comment]
I already have tried my Magento administrative credentials. I even reset my password to be sure. Still can't login.

**Comment by sarog** [comment]
I finally figured it out.

An additional line must be added to the .htaccess or Apache configuration file to have this method of authorization enabled:

`SetEnvIfNoCase ^Authorization$ "(.+)" HTTP_AUTHORIZATION=$1`

I added it in my VirtualHost configuration, right before my ProxyPassMatch:

```
    SetEnvIfNoCase ^Authorization$ "(.+)" HTTP_AUTHORIZATION=$1

    ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi://127.0.0.1:9000/usr/home/jcc/www/$1
```

Apache 2.4.18, php-fpm 5.6.19, Magento 1.9.2.4, using mod_proxy_fcgi on FreeBSD 10.2

**Comment by ts-** [comment]
Basically the problem is if you have CGI. That does not set the HTTP auth variables - they are always null. As such, if you have a database, it won't use the default magmi:magmi and will never verify. While you can do the above method, some have suggested .htaccess modifications etc, I prefer that things work out of the box as well, without server changes, especially for those who don't have the skillset for it.

If you open magmi/web/security.php you can comment it entirely out, though I prefer to wrap it all in:
`$sapiType = php_sapi_name();
if(substr($sapiType, 0, 3) != 'cgi'){
 ... authentication ... 
}`
This checks if you have CGI and if so, bypasses the HTTP auth.

In my personal view, I like Magmi to be separate from Magento admin (or have the option for different verification). For example we use password autofillers to log into Magento admin - the users don't actually have the password and can't copy it. So this method of authentication is not suitable for us. Though it might be good to note that two sets of authentication can offer security too - not just Magento admin users, but a set username and pass of people who are allowed to access Magmi - e.g. people who know what they are doing.

There is also one major security flag in using the Magento password - it does not check if the user is active!
